src/sentry/integrations/gitlab/client.py

```diff
@@ -176,5 +216,46 @@ def create_project_webhook(self, project_id):
         return resp['id']
 
     def delete_project_webhook(self, project_id, hook_id):
+        """Delete a webhook from a project
+
+        See https://docs.gitlab.com/ee/api/projects.html#delete-project-hook
+        """
         path = GitLabApiClientPath.project_hook.format(project=project_id, hook_id=hook_id)
         return self.delete(path)
+
+    def get_last_commits(self, project_id, end_sha):
+        """Get the last set of commits ending at end_sha
+
+        Gitlab doesn't give us a good way to do this, so we fetch the end_sha
+        and use its date to find the block of commits. We only fetch one page
+        of commits to match other implementations (github, bitbucket)
```

Comment:
By using this technique, is it possible to miss a commit that – occurs in the commit graph "after" `end_sha` (e.g. like a child node) – but is _dated_ in the past (e.g. by a weird merge, or someone re-writing history, etc)?

e.g. if this was the commit graph:

```
c (sep 30, HEAD)
|
b (oct 2)
|
a (oct 1, `end_sha`)
|
...
```

Reply:
Okay, just reviewed the GitLab docs, and I see that `created_at` is when the commit was created on GitLab, and I'm thinking about `authored_date`. Nevermind – _seems_ correct!